R/derive_vars_aage.R

```diff
@@ -123,22 +123,64 @@ NULL
 #' data.frame(AGE = 1:100) %>%
 #'   derive_agegr_fda(age_var = AGE, new_var = AGEGR1)
 #'
-derive_agegr_fda <- function(dataset, age_var, new_var) {
+derive_agegr_fda <- function(dataset, age_var, age_unit = NULL, new_var) {
 
   age_var <- assert_symbol(enquo(age_var))
   new_var <- assert_symbol(enquo(new_var))
   assert_data_frame(dataset, required_vars = quo_c(age_var))
 
-  out <- mutate(
-    dataset,
-    !!new_var := cut(
-      x = !!age_var,
-      breaks = c(0, 19, 65, Inf),
-      labels = c("<=18", "19-64", ">=65"),
-      include.lowest = TRUE,
-      right = FALSE
-    )
-  )
+  if (!paste0(quo_get_expr(age_var), "U") %in% colnames(dataset)) {
+
+    if (is.null(age_unit)) {
+
+      err_msg<- paste(
+        "There is no variable unit: ", paste0(quo_get_expr(age_var), "U") ,
+        "associated with ", quo_get_expr(age_var),
+        "and the argument `age_unit` is missing. Please specify a value for `age_unit`"
+      )
+      abort(err_msg)
+    } else{
+      assert_character_scalar(tolower(age_unit), values = valid_time_units())
+      ds <-dataset %>%
+        mutate(temp_age = time_length(duration(!!age_var, units = tolower(age_unit)), unit= "years"))
+    }
+  } else {
+
+    unit<- tolower(unique(pull(dataset,!!sym(paste0(quo_get_expr(age_var), "U")) )))
```

Comment:
That's a good point. We could either make it so it can handle any unit using a `case_when` statement or issue an error that multiple units are present. I think it should error when multiple units are present as this shouldn't be the case after the `derive_vars_aage` function has been used. What do you think?

Reply:
I think we can not assume that `derive_vars_aage` is called because `AAGE` is not a required variable.

For studies with adult subjects it should be unlikely to have a different unit than years but I am not sure what to expect for studies with pediatric subjects. As `AGEU` is a required variable, I would guess that multiple units can occur. Thus I would handle multiple units.